--- a/toolkit/crashreporter/breakpad-client/linux/handler/minidump_descriptor.h
+++ b/toolkit/crashreporter/breakpad-client/linux/handler/minidump_descriptor.h
@@ -31,6 +31,7 @@
 
 #include <assert.h>
 #include <sys/types.h>
+#include <cstdint>
 
 #include <string>
 
--- a/mfbt/tests/TestIntegerRange.cpp
+++ b/mfbt/tests/TestIntegerRange.cpp
@@ -7,6 +7,7 @@
 #include "mozilla/Assertions.h"
 #include "mozilla/IntegerRange.h"
 
+#include <cstdint>
 #include <stddef.h>
 
 using mozilla::IntegerRange;
--- a/mfbt/tests/TestArray.cpp
+++ b/mfbt/tests/TestArray.cpp
@@ -5,6 +5,7 @@
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "mozilla/Array.h"
+#include <cstdint>
 
 void TestInitialValueByConstructor() {
   using namespace mozilla;
--- a/mfbt/tests/TestNotNull.cpp
+++ b/mfbt/tests/TestNotNull.cpp
@@ -5,6 +5,7 @@
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include <type_traits>
+#include <cstdint>
 
 #include "mozilla/NotNull.h"
 #include "mozilla/RefPtr.h"
--- a/mfbt/tests/TestSmallPointerArray.cpp
+++ b/mfbt/tests/TestSmallPointerArray.cpp
@@ -5,6 +5,7 @@
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "mozilla/SmallPointerArray.h"
+#include <cstdint>
 
 #define PTR1 (void*)0x4
 #define PTR2 (void*)0x5
--- a/mfbt/tests/TestSPSCQueue.cpp
+++ b/mfbt/tests/TestSPSCQueue.cpp
@@ -7,6 +7,7 @@
 #include "mozilla/SPSCQueue.h"
 #include "mozilla/PodOperations.h"
 #include <vector>
+#include <cstdint>
 #include <iostream>
 #include <thread>
 #include <chrono>
